base_layer/core/tests/block_validation.rs

```diff
@@ -52,3 +66,206 @@ fn test_genesis_block() {
     let result = db.add_block(block);
     assert!(result.is_ok());
 }
+
+#[test]
+fn test_valid_chain() {
```

Comment:
Here's how I'd prefer to structure this test:

1. Setup blockchain with full validation.
2. Use the included testnet genesis block
3. use `generate_new_block` to build new block(s)
4. Query the DB and compare against expectations
5. Repeat 3, 4 as desired.

Essentially, don't fiddle with the blockchain logic; just let it do its thing.

And then riffing here -- I'm thinking of ways we can separate the test suite for CI  into 
1. always run - unit tests & very fast integration tests
2. run 3times per day - others
3. Daily? - really slow / tests involving mining etc

This would fall into the latter category